### DISCUSSION TITLE
AMD SMBus

### DISCUSSION THREAD

**Discussion by myszkin81** [liquidctl/liquidctl#302] [Ideas]
Are there any plans for supporting growing family of AMD users?
I tried my Vengance RAM but eventualy found out that currently only Intel SMBus is supported.
also is there a development roadmap available for this project?

**Comment by MarshallAsch** [comment]
@jonasmalacofilho If I remember correctly you asked me to test some stuff for ram detection on AMD, I had thought that the test you asked me to do worked after a different kernel module was loaded.

**Comment by MarshallAsch** [comment]
> also is there a development roadmap available for this project?

We do not currently have a development road map for this project we do have a couple of  [milestones](https://github.com/liquidctl/liquidctl/milestones) that get set that contain tasks that we aim to have completed for the upcoming release. But getting bigger tasks completed, especially ones that involve the SMBus interface without having the actual hardware to test and develop with has been a huge challenge.

**Comment by jonasmalacofilho** [comment]
> Are there any plans for supporting growing family of AMD users?

Yes, I would very much like to support SMBus on AMD boards (on Linux) as soon as possible.

The main issue right now is that the `i2c-piix4` kernel driver only appears to expose the first page of the SPD EEPROM, and we need the second page to check the manufacturer and model numbers, which we use during the device detection phase.

Besides that, as @MarshallAsch reminded, it might also be necessary to manually load `ee1004`, which isn't necessary with Intel/`i2c-i801`.

---

Can you execute and show the outputs of

```
$ lsmod
$ ls -l /sys/bus/i2c/devices
# modprobe i2c-dev 
# modprobe ee1004
$ ls -l /sys/bus/i2c/devices
```

**Reply by myszkin81** [reply]
@MarshallAsch 
>especially ones that involve the SMBus interface without having the actual hardware to test and develop

I can help as long as it doesn't involve programming. I have H100i Platinium SE, Commader Pro, Vengance Pro RGB Enchancement Kit


@jonasmalacofilho 
My lsmod is not quite a good one as I am using Gentoo and most modules are build in to the kernel

```
nvidia_drm             57344  9
nvidia_modeset       1179648  15 nvidia_drm
nvidia              33529856  605 nvidia_modeset
wmi_bmof               16384  0
uvcvideo              110592  0
videobuf2_vmalloc      16384  1 uvcvideo
videobuf2_memops       16384  1 videobuf2_vmalloc
snd_usb_audio         262144  0
videobuf2_v4l2         28672  1 uvcvideo
sp5100_tco             16384  0
videobuf2_common       53248  2 videobuf2_v4l2,uvcvideo
snd_usbmidi_lib        28672  1 snd_usb_audio
i2c_piix4              24576  0
i2c_nvidia_gpu         16384  0
mac_hid                16384  0
xhci_pci               20480  0
xhci_pci_renesas       16384  1 xhci_pci
```

ls -l /sys/bus/i2c/devices

```
total 0
lrwxrwxrwx 1 root root 0 Mar  3 23:13 0-0008 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.3/i2c-0/0-0008
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-0 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.3/i2c-0
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-1 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-1
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-2 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-2
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-3 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-3
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-4 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-4
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-5 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-5
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-6 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-6
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-7 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-7
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-8 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-8
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-9 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-9
```

after loading i2c-dev and ee1004 modules the output is  same

ls -l /sys/bus/i2c/devices
```
total 0
lrwxrwxrwx 1 root root 0 Mar  3 23:13 0-0008 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.3/i2c-0/0-0008
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-0 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.3/i2c-0
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-1 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-1
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-2 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-2
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-3 -> ../../../devices/pci0000:00/0000:00:14.0/i2c-3
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-4 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-4
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-5 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-5
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-6 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-6
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-7 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-7
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-8 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-8
lrwxrwxrwx 1 root root 0 Mar  3 23:12 i2c-9 -> ../../../devices/pci0000:00/0000:00:03.1/0000:05:00.0/i2c-9
```

**Reply by jonasmalacofilho** [reply]
Thanks!

> after loading i2c-dev and ee1004 modules the output is same

Hm, apparently I don't remember very well what @MarshallAsch and I tested.

I think if you manage to get `decode-dimms` to work (from i2c-tools), then that *probably* means you have all necessary kernel modules loaded.  And after that it's just a matter of someone getting liquidctl to read the SPD info just as well as `decode-dimms`.

> I can help as long as it doesn't involve programming

Unfortunately I personally don't have the bandwidth to work on something like this right now: without access to an AMD/PIIX4 board, even figuring out trivial stuff takes hours of reading kernel source code, instead of minutes exploring sysfs, and write-test cycles get super long.

> Vengance Pro RGB

FYI, we currently only support the non-pro Vengeance RGB protocol.
